src/databricks/sql/auth/thrift_http_client.py

```diff
@@ -28,6 +30,7 @@ def __init__(
         key_file=None,
         ssl_context=None,
         max_connections: int = 1,
+        retry_policy: Union[DatabricksRetryPolicy, int] = 0,
```

Comment:
Could we not accomplish the same thing with an Option?

Reply:
No. `Optional` means it can be either the specified type or `None` ([docs](https://docs.python.org/3/library/typing.html#typing.Optional)). In this case it must be either `0` (urllib3 retries nothing) or `DatabricksRetryPolicy`, so `Union` is the correct route.